chain/types/ethtypes/eth_1559_transactions.go

```diff
@@ -0,0 +1,384 @@
+	if err != nil {
+		return address.Undef, err
+	}
+
+	return ea.ToFilecoinAddress()
+}
+
+func (tx *Eth1559TxArgs) ToVerifiableSignature(sig []byte) ([]byte, error) {
+	return sig, nil
+}
+
+func (tx *Eth1559TxArgs) ToEthTx(smsg *types.SignedMessage) (EthTx, error) {
+	from, err := EthAddressFromFilecoinAddress(smsg.Message.From)
+	if err != nil {
+		return EthTx{}, xerrors.Errorf("sender was not an eth account")
+	}
+	hash, err := tx.TxHash()
+	if err != nil {
+		return EthTx{}, err
+	}
+	gasFeeCap := EthBigInt(tx.MaxFeePerGas)
+	gasPremium := EthBigInt(tx.MaxPriorityFeePerGas)
+
+	ethTx := EthTx{
+		ChainID:              EthUint64(build.Eip155ChainId),
+		Type:                 Eip1559TxType,
+		Nonce:                EthUint64(tx.Nonce),
+		Hash:                 hash,
+		To:                   tx.To,
+		Value:                EthBigInt(tx.Value),
+		Input:                tx.Input,
+		Gas:                  EthUint64(tx.GasLimit),
+		MaxFeePerGas:         &gasFeeCap,
+		MaxPriorityFeePerGas: &gasPremium,
+		From:                 from,
+		R:                    EthBigInt(tx.R),
+		S:                    EthBigInt(tx.S),
+		V:                    EthBigInt(tx.V),
+	}
+
+	return ethTx, nil
+}
+
+func (tx *Eth1559TxArgs) InitialiseSignature(sig typescrypto.Signature) error {
+	if sig.Type != typescrypto.SigTypeDelegated {
+		return xerrors.Errorf("RecoverSignature only supports Delegated signature")
+	}
+
+	if len(sig.Data) != 65 {
+		return xerrors.Errorf("signature should be 65 bytes long, but got %d bytes", len(sig.Data))
+	}
+
+	r_, err := parseBigInt(sig.Data[0:32])
+	if err != nil {
+		return xerrors.Errorf("cannot parse r into EthBigInt")
+	}
+
+	s_, err := parseBigInt(sig.Data[32:64])
+	if err != nil {
+		return xerrors.Errorf("cannot parse s into EthBigInt")
+	}
+
+	v_, err := parseBigInt([]byte{sig.Data[64]})
+	if err != nil {
+		return xerrors.Errorf("cannot parse v into EthBigInt")
+	}
+
+	tx.R = r_
+	tx.S = s_
+	tx.V = v_
+
+	return nil
+}
+
+func (tx *Eth1559TxArgs) packTxFields() ([]interface{}, error) {
+	chainId, err := formatInt(tx.ChainID)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce, err := formatInt(tx.Nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	maxPriorityFeePerGas, err := formatBigInt(tx.MaxPriorityFeePerGas)
+	if err != nil {
+		return nil, err
+	}
+
+	maxFeePerGas, err := formatBigInt(tx.MaxFeePerGas)
+	if err != nil {
+		return nil, err
+	}
+
+	gasLimit, err := formatInt(tx.GasLimit)
+	if err != nil {
+		return nil, err
+	}
+
+	value, err := formatBigInt(tx.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	res := []interface{}{
+		chainId,
+		nonce,
+		maxPriorityFeePerGas,
+		maxFeePerGas,
+		gasLimit,
+		formatEthAddr(tx.To),
+		value,
+		tx.Input,
+		[]interface{}{}, // access list
+	}
+	return res, nil
+}
+
+func parseEip1559Tx(data []byte) (*Eth1559TxArgs, error) {
+	if data[0] != Eip1559TxType {
+		return nil, xerrors.Errorf("not an EIP-1559 transaction: first byte is not 2")
```

Comment:
```suggestion
		return nil, xerrors.Errorf(fmt.Sprintf("not an EIP-1559 transaction: first byte is not %d", Eip1559TxType))
```

not important, just reducing magic

Reply:
or even `0x%x` perhaps